wp-includes/widgets/class-wp-widget-media.php

```diff
@@ -234,6 +235,45 @@ class="media-widget-instance-property"
 	}
 
 	/**
+	 * Deletes the Widget when the attachment file is deleted.
+	 *
+	 * @since 4.8.0
+	 * @access public
+	 *
+	 * @param int $post_id Attachment ID.
+	 */
+	public function delete_attachment_data( $post_id ) {
```

Comment:
➕  I was thinking the same thing - when the media item is deleted, perhaps just not render anything - or perhaps if the user has a certain role, and is viewing the site, then show a notice in the widget area that the attachment is missing.